pkg/queue/worker.go

```diff
@@ -52,43 +66,29 @@ func (w *worker) Stop() {
 	log.Trace("Job queue shutdown completed")
 }
 
+// workerContext is a context for running scan jobs.
 type workerContext struct {
+	controller scan.Controller
 }
 
+// ScanArtifact is a handler function for the specified scan Job with the given workerContext.
 func (s *workerContext) ScanArtifact(job *work.Job) (err error) {
 	log.WithField("scan_job_id", job.ID).Debug("Executing enqueued scan job")
 
-	controller, err := s.controller()
```

Comment:
Here was the actual bug, where we called the controller method, which in turn, instantiated new Redis store, which in turn created a new connection Pool.
By injecting the scan.Controller we do reuse store instance, and in turn, the underlying Redis connection pool.
This is a prerequisite for supporting Redis Sentinel.